Propagate purity information for member access to foreign pure variables (#12928)
ethereum/solidity: Changelog.md

Compiler Features:
* LSP: Add rudimentary support for semantic highlighting.

> * TypeChecker: Support using foreign constants as compile-time constants

Reviewer: I think this wording might be a little clearer to users:
```suggestion
* TypeChecker: Support using library constants in initializers of other constants.
```

The change seems more generic but looks like in practice it only affected library constants.

Author: Agree, done.

Compiler Features:
 * LSP: Add rudimentary support for semantic highlighting.
 * Yul Optimizer: Improve inlining heuristics for via IR code generation and pure Yul compilation.

>  * TypeChecker: Support using library constants in initializers of other constants

Reviewer: @nishant-sachdeva Now that we released 0.8.15 the PR needs to be rebased and this moved to 0.8.16 changelog.
